internal/rbd/rbd_util.go

```diff
@@ -193,28 +203,94 @@ type migrationVolID struct {
 	clusterID string
 }
 
-var supportedFeatures = map[string]imageFeature{
-	librbd.FeatureNameLayering: {
-		needRbdNbd: false,
-	},
-	librbd.FeatureNameExclusiveLock: {
-		needRbdNbd: false,
-	},
-	librbd.FeatureNameObjectMap: {
-		needRbdNbd: false,
-		dependsOn:  []string{librbd.FeatureNameExclusiveLock},
-	},
-	librbd.FeatureNameFastDiff: {
-		needRbdNbd: false,
-		dependsOn:  []string{librbd.FeatureNameObjectMap},
-	},
-	librbd.FeatureNameJournaling: {
-		needRbdNbd: true,
-		dependsOn:  []string{librbd.FeatureNameExclusiveLock},
-	},
-	librbd.FeatureNameDeepFlatten: {
-		needRbdNbd: false,
-	},
+var (
+	supportedFeatures = map[string]imageFeature{
+		librbd.FeatureNameLayering: {
+			needRbdNbd: false,
+		},
+		librbd.FeatureNameExclusiveLock: {
+			needRbdNbd: false,
+		},
+		librbd.FeatureNameObjectMap: {
+			needRbdNbd: false,
+			dependsOn:  []string{librbd.FeatureNameExclusiveLock},
+		},
+		librbd.FeatureNameFastDiff: {
+			needRbdNbd: false,
+			dependsOn:  []string{librbd.FeatureNameObjectMap},
+		},
+		librbd.FeatureNameJournaling: {
+			needRbdNbd: true,
+			dependsOn:  []string{librbd.FeatureNameExclusiveLock},
+		},
+		librbd.FeatureNameDeepFlatten: {
+			needRbdNbd: false,
+		},
+	}
+
+	krbdLayeringSupport = []util.KernelVersion{
+		{
+			Version:      3,
+			PatchLevel:   8,
+			SubLevel:     0,
+			ExtraVersion: 0,
+			Distribution: "",
+			Backport:     false,
+		},
+	}
+	krbdStripingV2Support = []util.KernelVersion{
+		{
+			Version:      3,
+			PatchLevel:   10,
+			SubLevel:     0,
+			ExtraVersion: 0,
+			Distribution: "",
+			Backport:     false,
+		},
+	}
+	krbdExclusiveLockSupport = []util.KernelVersion{
+		{
+			Version:      4,
+			PatchLevel:   9,
+			SubLevel:     0,
+			ExtraVersion: 0,
+			Distribution: "",
+			Backport:     false,
+		},
+	}
+	krbdDataPoolSupport = []util.KernelVersion{
+		{
+			Version:      4,
+			PatchLevel:   11,
+			SubLevel:     0,
+			ExtraVersion: 0,
+			Distribution: "",
+			Backport:     false,
+		},
+	}
+)
+
+// prepareKrbdFeatureAttrs prepare krbd fearure set based on kernel version.
+// Minimum kernel version should be 3.6, else it will return error.
+func prepareKrbdFeatureAttrs() (string, error) {
+	// fetch the current running kernel info
+	release, err := util.GetKernelVersion()
+	if err != nil {
+		return "", fmt.Errorf("fetching current kernel version failed: %w", err)
+	}
+
+	if util.CheckKernelSupport(release, krbdDataPoolSupport) {
+		return strconv.FormatInt(rbdDataPoolSupportAttr, 16), nil
+	} else if util.CheckKernelSupport(release, krbdExclusiveLockSupport) {
+		return strconv.FormatInt(rbdExclusiveLockSupportAttr, 16), nil
+	} else if util.CheckKernelSupport(release, krbdStripingV2Support) {
+		return strconv.FormatInt(rbdStripingV2SupportAttr, 16), nil
+	} else if util.CheckKernelSupport(release, krbdLayeringSupport) {
+		return strconv.FormatInt(rbdLayeringSupportAttr, 16), nil
+	}
```

Comment:
@Madhu-1 I don't see how to convert this if-Else to Switch as suggested by linter. Can I just add a nolint here? thoughts?

Reply:
Yes should be fine :+1:  Please add a reason for adding nolint.

Reply:
Sure. Thanks!